### PR TITLE
feat: expose md5 function

### PR DESCRIPTION
**Pull request by kneekey23**
*Description of changes:* Exposes md5 function

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by justinboswell — APPROVED** [review]

File: Source/AwsCommonRuntimeKit/Utilities.swift [hunk 1]
```diff
@@ -25,6 +26,22 @@ extension String {
...
+        let buffer = aws_byte_buf(len: 0, buffer: bufferPtr, capacity: 16, allocator: allocator.rawValue)
+        let output: UnsafeMutablePointer<aws_byte_buf> = fromPointer(ptr: buffer)
+
+        let result = aws_md5_compute(allocator.rawValue, input, output, truncate)
+        if result == AWS_OP_SUCCESS {
+            let byteCursor = aws_byte_cursor_from_buf(output)
+            return byteCursor.toData().map { String(format: "%02hhx", $0) }.joined()
+        } else {
+            return nil
+        }
```
[the remark this answers is not in the document]
Comment (kneekey23) [reply Source/AwsCommonRuntimeKit/Utilities.swift]: updated

File: Source/AwsCommonRuntimeKit/Utilities.swift [hunk 2]
```diff
@@ -25,6 +26,21 @@ extension String {
     var awsByteCursor: aws_byte_cursor {
         return aws_byte_cursor_from_c_str(self.asCStr())
     }
+
+    func md5(allocator: Allocator = defaultAllocator, truncate: Int = 0) -> String? {
```
Comment (wooj2) [inline Source/AwsCommonRuntimeKit/Utilities.swift:30]: Maybe it would be good to change the name of this function to:
```
base64EncodedMD5
```

Comment (wooj2) [reply Source/AwsCommonRuntimeKit/Utilities.swift:30]: also needs the `public`modifier

Comment (kneekey23) [reply Source/AwsCommonRuntimeKit/Utilities.swift]: updated

**Review by wooj2 — APPROVED** [review]
lgtm!!

**Pull request merged**